### ISSUE TITLE
Scripts vs ERB files

### ISSUE THREAD

**Issue by gerhard** [emalm/tmux-boshrelease#2]
I prefer to keep ERB files separate from scripts so that [shellcheck](https://github.com/koalaman/shellcheck) can do its magic.

[What are your thoughts for making tmux available to the shell profile like this?]( https://github.com/rabbitmq/rabbitmq-server-boshrelease/blob/master/jobs/rabbitmq-server/templates/bin/pre-start#L53-L57)

It's subtle, but [referencing job packages](https://github.com/rabbitmq/rabbitmq-server-boshrelease/blob/master/jobs/rabbitmq-server/templates/env.erb#L20) instead of global packages is [a good pattern](https://groups.google.com/a/pivotal.io/d/msg/cf-eng/KHl7soQdtFo/-cot5qLYBQAJ).

**Comment by emalm** [comment]
Thanks for the suggestions, @gerhard! I think you're proposing three separate changes:

1. Refer exclusively to job-specific packages.
1. Reorganize and rename the template files to keep the BOSH lifecycle scripts separate from the other files, as well as to isolate ERB content.
1. Make the profile.d script a pointer to a job template file that performs the environmental setup, instead of doing it inline.

Point 1 makes sense for sure, and I'll push out that change shortly. For point 2, I can see the benefits of that organizational separation, as well as separating ERB content from any complicated scripting logic. Technically, there is no non-trivial ERB content in the release so far, and BOSH always renders all the templates regardless of whether they have an `.erb` extension, so I think I'll remove the extension from all of the files in the templates directory.

For point 3, I'm not sure I see the value of the indirection in this case, since this release is currently so simple. If it gets to the point where several different scripts would need the same environmental setup, as seems to the case in [the RabbitMQ release](https://github.com/rabbitmq/rabbitmq-server-boshrelease), I think then it would justify the indirection. Could you explain why you think it would be beneficial in this case?

Thanks,
Eric

**Comment by gerhard** [comment]
Actually, I was thinking that you might want to write the file directly. `pre-start` would just become:

```sh
#!/usr/bin/env bash

cat > /etc/profile.d/add_tmux_to_path.sh <<EOF
if [[ "\$PATH" != *"/var/vcap/jobs/tmux/packages/tmux/bin"* ]]; then
  export PATH="/var/vcap/jobs/tmux/packages/tmux/bin:\$PATH"
fi
EOF
```

As for `/var/vcap/jobs/*/env`, I wish it became a BOSH standard since it makes it easier to reason about job-specific environment variables. I see this as stepping stone towards just-in-timing variables, via CredHub or similar. I'm sure @cppforlife has other/more ideas on the matter.

**Comment by gerhard** [comment]
I'm happy to conclude this with 1. & 2.

Point 3. has more nuance and feels more of a preference than good practice.

Thanks!

**Comment by emalm** [comment]
Thanks again, @gerhard ! I published [v0.1.0](https://github.com/emalm/tmux-boshrelease/releases/tag/v0.1.0) earlier today with those changes.

**Comment by gerhard** [comment]
👍
